python/cuspatial/cuspatial/core/spatial/nearest_points.py

```diff
@@ -0,0 +1,107 @@
+import cupy as cp
+
+import cuspatial._lib.nearest_points as nearest_points
+from cuspatial.core._column.geocolumn import GeoColumn
+from cuspatial.core.geodataframe import GeoDataFrame
+from cuspatial.core.geoseries import GeoSeries
+from cuspatial.utils.column_utils import (
+    contains_only_linestrings,
+    contains_only_points,
+)
+
+
+def pairwise_point_linestring_nearest_points(
+    points: GeoSeries, linestrings: GeoSeries
+) -> GeoDataFrame:
+    """Returns the nearest points between two GeoSeries of points and
+    linestrings.
+
+    Parameters
+    ----------
+    points : GeoSeries
+        A GeoSeries of points. Currently either only a series of points or
+        a series of multi-points is supported.
+    linestrings : GeoSeries
+        A GeoSeries of linestrings.
```

Comment:
What about multilinestrings?

Reply:
Also supported. cuspatial distinguishes points from multipoints but not linestrings with multilinestrings. All linestrings are stored as multlinestrings under the hood. Which is why we support the mixture of them.